servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/AsyncContextProvider.java

```diff
@@ -47,6 +47,12 @@ interface AsyncContextProvider {
      */
     ContextMap context();
 
+    /**
+     * Set the current {@link ContextMap}
+     * @param contextMap the {@link ContextMap} to set in the thread local state.
+     */
+    void setContextMap(ContextMap contextMap);
```

Comment:
Then it looks more like `ContextMapHolder`, should we just make this interface extend the holder?

Reply:
I did do it, but it's a bit itchy due to the `@Nullable` annotations and the return type of `.context(ContextMap)`.